### ISSUE TITLE
Artifact Power tooltip may overlap GameTooltip

### ISSUE THREAD

**Issue by lotheac** [AzeriteTeam/AzeriteUI#39]
Found this by opening bags and accidentally mousing over the artifact power indicator when trying to look at something in my bags. When moving the cursor over the bag frame, the AP tooltip does not disappear, causing possible overlap.

![tooltip-overlap](https://user-images.githubusercontent.com/167940/60910536-bc9aed80-a289-11e9-8721-7c7aef2cc04a.jpg)

**Comment by goldpaw** [comment]
You have an amazing talent for discovering the weirdest of bugs! :smile: 

Anyway, which tooltip appeared first? My tooltip library attempts to hide the GameTooltip when any of my tooltips are shown, so I would assume this happens if you had the mouse above the artifact toggle button, and then opened the bags? 

Technically this shouldn't happen, because the OnLeave script should've fired for one of them, but obviously it didn't. 

I'll look into it!

**Comment by lotheac** [comment]
On Tue, Jul 09 2019 11:19:38 -0700, Lars Norberg wrote:
> Anyway, which tooltip appeared first? My tooltip library attempts to hide the GameTooltip when any of my tooltips are shown, so I would assume this happens if you had the mouse above the artifact toggle button, and then opened the bags? 

Actually I already had the bags open before mousing over anything. Just
moving the cursor over the AP indicator, and then over the bags frame,
the AP tooltip does not disappear. So the AP tooltip was first.

-- 
Lauri Tirkkonen | lotheac @ IRCnet

**Comment by goldpaw** [comment]
That's just so strange. The OnLeave script should automatically kick in when another OnEnter script fires. 

I have some ideas for fixes, just gotta figure out more exact what's happening first. I'll get online and try this myself. :+1:

**Comment by goldpaw** [comment]
Found the bug, and can reproduce: 

1. Touch the toggle button
2. Make sure the ring frame is visible
3. Move directly to something that triggers the GameTooltip, but also is above the ringframe. 
4. Chaos. 

This happens because the togglebutton's OnLeave script doesn't hide the azerite tooltip if the mouse pointer is above the ringframe. Turns out the API call I'm using to check where the pointer is at, `MouseIsOver(frame)` doesn't care if something is covering the frame in question, nor even if the frame is visible at all. As long as the mouse is where the frame is or would be, then it returns true. Thus the chaos. 

So I'm fixing this by just adding this nifty little local function at the start, that makes our own version of the API call, that works the way we need it to; 

```Lua
local MouseIsOver = function(frame)
	return (frame == GetMouseFocus())
end
```
That fixes it, without altering anything else in the code. Yay! XD

**Comment by goldpaw** [comment]
Fixed in 1.2.158-Release.
